File: app/src/main/java/net/north101/android/ghplayertracker/data/PerkAction.java
package net.north101.android.ghplayertracker.data;

import org.json.JSONException;

public enum PerkAction {
    add,
    remove;

    public static PerkAction parse(String perkActionData) throws JSONException {
        switch (perkActionData) {
            case "add":
                return PerkAction.add;

            case "remove":
                return PerkAction.remove;

            default:
                throw new JSONException("");
        }
    }
}

File: app/src/main/java/net/north101/android/ghplayertracker/data/PerkAction.kt
package net.north101.android.ghplayertracker.data

import org.json.JSONException

enum class PerkAction {
    add,
    remove;

    companion object {
        @Throws(JSONException::class)
        fun parse(perkActionData: String): PerkAction {
            return when (perkActionData) {
                "add" -> PerkAction.add
                "remove" -> PerkAction.remove
                else -> throw JSONException(perkActionData)
            }
        }
    }
}
